src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/InteractiveMapsMercuryComponentObject.java

```diff
@@ -82,9 +90,16 @@ public void clickZoomOut(int amountOfClicks) {
   }
 
   public void clickPin() {
-    waitForElementVisibleByElement(poiPin);
-    tapOnElement(poiPin);
-    PageObjectLogging.log("clickPin", "Pin was clicked", true);
+    String methodName = "clickPin";
+    try {
+      driver.switchTo().frame(mapFrame);
+      waitForElementVisibleByElement(poiPin);
+      tapOnElement(poiPin);
+      PageObjectLogging.log(methodName, "Pin was clicked", true);
```

Comment:
Move this outside of try/catch as this method would not throw the error that is expecting in the catch block